### DISCUSSION TITLE
Alternative programmers

### DISCUSSION THREAD

**Discussion by ConorIA** [rbaron/b-parasite#86] [Q&A]
It seems like the J-LINK EDU Mini are discontinued on [Adafruit](https://www.adafruit.com/product/3571) and [Sparkfun](https://www.sparkfun.com/products/retired/15345) are now retired. I am looking for alternatives, and thought it would be worth seeking recommendations. 

So far, a Google Shopping search had led me to this [Seggar branded debugger for ~$20](https://www.aliexpress.us/item/3256803831857338.html?pdp_npi=2%40dis%21USD%21US%20%2417.28%21%2412.10%21%21%21%21%21%40210312eb16712908052273996e28e6%2112000027754310653%21btf&_t=pvid%3Addbca341-3338-49e3-bb68-9219e1641449&afTraceInfo=1005004018172090__pc__pcBridgePPC__xxxxxx__1671290805&spm=a2g0o.ppclist.product.mainProduct&gatewayAdapt=glo2usa&_randl_shipto=US). That price strikes me as a little suspicious, but I really don't know. Then there are several sellers of some generic clone board ([1](https://www.aliexpress.us/item/3256801639330842.html?pdp_npi=2%40dis%21USD%21US%20%246.70%21%245.29%21%21%21%21%21%40210312eb16712908871554902e28e6%2112000017754018195%21btf&_t=pvid%3A8676a4ff-0dd9-4b5e-8341-72935c5c5e73&afTraceInfo=1005001825645594__pc__pcBridgePPC__xxxxxx__1671290887&spm=a2g0o.ppclist.product.mainProduct&gatewayAdapt=glo2usa&_randl_shipto=US), [2](https://www.aliexpress.us/item/3256804645201659.html?pdp_npi=2%40dis%21USD%21US%20%247.04%21%244.93%21%21%21%21%21%40210312eb16712908973995006e28e6%2112000030651808275%21btf&_t=pvid%3A78fb15e7-192b-44c4-b684-555ac9f99c60&afTraceInfo=1005004831516411__pc__pcBridgePPC__xxxxxx__1671290897&spm=a2g0o.ppclist.product.mainProduct&gatewayAdapt=glo2usa&_randl_shipto=US), [3](https://www.aliexpress.us/item/3256803209803801.html?spm=a2g0o.productlist.main.1.4dc911e7dnhVlt&algo_pvid=e79179d3-b5d4-4aa0-9fb7-b74cfb23c303&aem_p4p_detail=202212170704041544597521980240003312006&algo_exp_id=e79179d3-b5d4-4aa0-9fb7-b74cfb23c303-0&pdp_ext_f=%7B%22sku_id%22%3A%2212000025590797209%22%7D&pdp_npi=2%40dis%21USD%216.9%216.9%21%21%21%21%21%40211be17616712894440894305d077a%2112000025590797209%21sea&curPageLogUid=WaZOzCTz0uY6&ad_pvid=202212170704041544597521980240003312006_1&ad_pvid=202212170704041544597521980240003312006_1&gatewayAdapt=4itemAdapt)), which are all around $10 with shipping. 

Any better suggestions? I expect that I would _only_ use this device for programming b-parasite boards, so cheaper is better!

**Comment by 1technophile** [comment]
See https://github.com/rbaron/b-parasite/issues/43

**Reply by ConorIA** [reply]
Thanks! It hadn't occured to me to search the issues.

**Reply by 1technophile** [reply]
I used the Black Magic Probe and it worked well

**Comment by drspangle** [comment]
Get a BluePill and follow the instructions in [#43](https://github.com/rbaron/b-parasite/issues/43#issuecomment-1336617818). It works flawlessly. Strongly recommend you get [the one from Amazon](https://www.amazon.com/gp/product/B079B7FGRB/) as it's from a trustworthy supplier and is cheap.

**Reply by ConorIA** [reply]
Thanks for the suggestion. Unfortunately that link becomes >C$100 product when it is shipped to Canada! I'll look for another source of the same package.

**Comment by dlavey** [comment]
I know this is an old thread, but in case someone finds it helpful in the future. I picked up one of [these](https://www.amazon.com/gp/product/B07V2CKG19/) and it's worked great for me, I'm not able to flash via Visual Studio Code, but I'm guessing it needs some tweaking and it would work. I use VS Code to compile and then nrfjprog to flash the device. I figured for $10, it was worth the risk to try it out. The little blue LED on it is brighter than the sun, but it has worked well for me.

**Reply by createordie1** [reply]
It`s been a while since someone was active here but could you provide a walkthrough?
I am trying the same but no success so far.

**Comment by nikosch86** [comment]
The J-Link Edu Mini is back in stock on adafruit. (https://www.adafruit.com/product/3571)
I was able to get mine directly from Segger in Germany though, very nice people :)
